alibi/saving.py

```diff
@@ -169,43 +169,60 @@ def _save_AnchorImage(explainer: 'AnchorImage', path: Union[str, os.PathLike]) -
 
 
 def _load_AnchorText(path: Union[str, os.PathLike], predictor: Callable, meta: dict) -> 'AnchorText':
-    # load the spacy model
-    import spacy
-    nlp = spacy.load(Path(path, 'nlp'))
+    from alibi.explainers import AnchorText
 
+    # load explainer
     with open(Path(path, 'explainer.dill'), 'rb') as f:
         explainer = dill.load(f)
 
-    explainer.nlp = nlp
+    perturb_opts = explainer.perturb_opts
+    sampling_method = explainer.sampling_method
+    nlp_sampling = [AnchorText.SAMPLING_UNKNOWN, AnchorText.SAMPLING_SIMILARITY]
 
-    # explainer._synonyms_generator contains spacy Lexemes which contain unserializable Cython constructs
-    # so we re-initialize the object here
-    # TODO: this is slow to re-initialize, try optimzing
-    from alibi.explainers.anchor_text import Neighbors
-    explainer._synonyms_generator = Neighbors(nlp_obj=nlp)
-    explainer.reset_predictor(predictor)
+    if sampling_method in nlp_sampling:
+        # load the spacy model
+        import spacy
+        model = spacy.load(Path(path, 'nlp'))
+    else:
+        # load language model
+        import alibi.utils.lang_model as lang_model
+        model_class = explainer.model_class
+        model = getattr(lang_model, model_class)(preloading=False)
+        model.from_disk(Path(path, 'language_model'))
```

Comment:
See comment in `lang_model.py`, not sure how this works without the class being initialized?

Reply:
In the last commit, the `LanguageModel` class has an extra flag `preloading` meaning whether to initialize the transformer with the online version or not. If the `preloading=False`, then a call from `from_disk` is expected. That's how it is done in the loading functionality. Something like this:
```python
model = DistilbertBaseUncased(preloading=False)
model.from_disk(path)
```